### ISSUE TITLE
Very slow to unlock or display UI + list of items when the self-hosted server is turned off

### ISSUE THREAD

**Issue by rchomat** [bitwarden/clients#9561]
### Steps To Reproduce

1. Turn off the self-hosted server
2. On the browser app, unlock the safe. The app stays on the unlock screen for a while. Multi click on the unlock button doesn't change anything, it's stuck during several minutes. If you close the app and reopen it, the app is just a grey background without anything it in for several minutes

### Expected Result

When a self-hosted server is disconnected, we should be able to use the local data without having to wait several minutes

### Actual Result

UI & items are shown several minutes after the app is opened or the safe unlocked

### Additional Context

I use to turn on my server only when I need to synchronise my clients or save a new password. The rest of the time it's turned off.

### Operating System

macOS

### Operating System Version

14.4.1

### Web Browser

Chrome

### Browser Version

125.0.6422.142

### Build Version

2024.5.2

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by maljac** [comment]
I experience the same issue:

Operating System
Windows 11 Pro

Operating System Version
22631.3737

Web Browser
Chrome

Browser Version
125.0.6422.142

Build Version
2024.5.2

**Comment by PylotLight** [comment]
Same issue

**Comment by popy2k14** [comment]
Here are also infos of this issue: https://github.com/bitwarden/clients/issues/10128
ALso happens with the desktop app.

**Comment by CirnoT** [comment]
Can confirm this still exists and is particularly visible on unstable networks when HTTP requests end up getting timeout after ~15s which causes entire UI to be more or less unusable.

This annoys me so much that I have decided to attempt to dig as to reason for why this happens, and while I was unable to dig too deep without compiling extension myself, I can confirm that the issue started sometimes in 2024 (most likely between 2024.02 and 2024.05.1, so that's the range one should bisect) and that the following are my observations:
- Extension always attempts to connect to provided identity server (usually `identity.bitwarden.com/connect/token`) to fetch new token based by stored refresh token
- The source of request is unknown, stack trace ends at `abstract-chrome-storage-api.service.ts`, however that request is routed via zone-ified `Promise` in order to retrieve value of `user_{}_token_refreshToken` which then gets used in actual request.
- Whoever is owner of this request explicitly wants request to happen synchronically and locks entire extension awaiting result from it
- The request is fired from `api.service.ts::nativeFetch` which calls zone-ified `fetch` providing standard browser Fetch API. Nowhere in this stack anything sets any timeout, so the TCP timeout for this HTTP request is entirely up to networking stack of browser and OS
- Replication is iffy at best; the extension seems to cache token for some time, so unlocking vault while networking is up and then locking it up and ensuring network replicates fault condition will not trigger issue as extension does not send request that causes this issue. Data is cached even after browser restart!

**Comment by gdzevo** [comment]
Same here, on a desktop app as well as with extension.
Offline mode is completely broken, the only workaround is to route `*.bitwarden.com` to `127.0.0.1`

**Comment by tombogle** [comment]
I'm not sure this is just a self-hosted issue. I've been getting it intermittently for the past couple months. Haven't paid careful attention, but now I'm thinking that maybe it only happens at work, so maybe it is environment/network related. I've found that if I disable the extension and then immediately re-enable it, the problem goes away (for a while).

**Comment by CirnoT** [comment]
> I'm not sure this is just a self-hosted issue.

It's not, in my example above I have been testing it by emulating severe packet loss to Bitwarden's cloud endpoints. It can certainly be more visible in self-hosted environments as chance for some connectivity issue there is higher. On the other hand, should some global outage happen to Bitwarden's cloud instances then we would also observe swarm of people reporting this issue (not everyone though, at least everyone that unlocked extension in past 24h should be fine).

**Comment by MatthiasGer0821** [comment]
same here.

**Comment by popy2k14** [comment]
@NovaSilentium Any news on this. Currently both (Browser Extension and Desktop Client (Windows in my case) are literally unusable for about ~30secs when self hosted server is not reachable.

**Comment by rchomat** [comment]
@justindbaur I've updated the plugin with the 2024.8.1 and the issue is still the same. Could you please reopen this issue and please take a look at the comment I wrote on your PR: https://github.com/bitwarden/clients/pull/10452#issuecomment-2303051315

**Comment by rchomat** [comment]
cc @trmartin4

**Comment by gdzevo** [comment]
same on firefox, issue is still present

**Comment by popy2k14** [comment]
issue is also still here on brave (chromium) browser and desktop client with latest 2024.8.1

**Comment by djsmith85** [comment]
The changes merged by @justindbaur have not been included in a release yet. They will be included in a following release.

**Comment by jason84019020** [comment]
The problem with the Chrome extension in version 2024.8.1 still persists. Despite the recent update, the issue has not been resolved.

**Comment by popy2k14** [comment]
@jason84019020 As @djsmith85 stated, the fix has not been released yet.

I hope it will be soon.

**Comment by MitosisByProxy** [comment]
Issue still persists as of version 2024.8.1 for firefox. Is there any indication in which version this will be included?

**Comment by popy2k14** [comment]
Yes, it's really unusable :-(

Please release this fix!!!!

**Comment by rchomat** [comment]
@justindbaur is it possible to release your fix as a hotfix ASAP ? The bug completely breaks the app since several month now !
That's unusable unless you wait several minutes to get access to your vault !

**Comment by cyberfox1** [comment]
> The changes merged by @justindbaur have not been included in a release yet. They will be included in a following release.

How could this not be fixed already???

**Comment by andysmile1** [comment]
Same here.  IOS app cannot load vault for around 60 seconds if the self-hosted server is offline.  The only way to load the vault instantly offline is that to turn my phone into airplane mode.

**Comment by p-rintz** [comment]
So, it seems kind of fixed in 2024.9.0.
This is for the Chrome extension. I wasnt able to test the desktop client or Firefox extension yet.

When unlocking while not connected to the Bitwarden instance, you can access the vault immediately afterwards.
But not until closing the window and reopening the extension.
If you input your password to unlock the vault, without a connection to the instance, the password input screen stays frozen for 30-60 seconds and does not transition to the vault, like it does when connected to the instance.

You can see this here. The vault is open (as shown with the 9+ in the BW icon), but the password screen stays.
![image](https://github.com/user-attachments/assets/22bb5980-bd32-433c-b1ce-fea3f4f2446f)

**Comment by rchomat** [comment]
@justindbaur @trmartin4

I've tried on chrome with the extension updated to the 2024.9.0 version and the behaviour is still the same issue. While trying to unlock the vault or when already unlocked and trying to access to it.

Please reopen this issue and do a HF in order ASAP we don't need to wait 3 other months with this very annoying bug

please take a look at my comment https://github.com/bitwarden/clients/pull/10452#issuecomment-2303051315

![Screenshot 2024-09-12 at 09 51 17](https://github.com/user-attachments/assets/e59f4bc4-b5e6-4247-b146-5f813627a20f)

**Comment by popy2k14** [comment]
Issue also not solved for me on newest brave (chromium based).
When accessing the vault with the extension it just shows nothing for about 1 minute:

![image](https://github.com/user-attachments/assets/2a776f18-9b9e-4d1e-97dc-92a129726464)

**Comment by neeqkowlah** [comment]
not solved for me with Version 2024.9.0 and Chrome Version 128.0.6613.119 (Build officiel) (64 bits)

**Comment by justindbaur** [comment]
2024.9.0 did not include the fix, it was itself a hotfix, we have a release coming up pretty soon that will include the fix. 

I can see the `/connect/token` calls taking awhile but those should largely be being done in the background and much more rarely. It's also code that hasn't changed in quite a while and we haven't previously had issues with it in this way, but a more recent change is having a feature flag to decide whether or not to show our upcoming UI refresh, that happens loads of times throughout the app and any delays there can have a noticable impact.

**Comment by rchomat** [comment]
@justindbaur why isn't it already pushed as a HF ? That's really boring and we're facing this issue since more than three months now.

**Comment by justindbaur** [comment]
@rchomat We have pretty strict rules around what makes something hotfix-worthy and about what can be included. This doesn't meet the standards we have.

**Comment by rchomat** [comment]
@justindbaur seems weird to me that something breaking a basic usage of the app is not considered as candidate for a HF but ok. 
When can we expect to see that fix deployed in order we don't bother you at each version ?

**Comment by MelHiour** [comment]
It's time to find some alternatives, I am afraid...

**Comment by victornavorskie** [comment]
same problem here, i have been using Bitwarden since 4 years and since 3 months is unusable and still no fix i think i will find something else

**Comment by kleinarne** [comment]
I'm not going to try and throw my weight around and declare this a reason to leave Bitwarden. But I do agree that this is an extremely frustrating issue, because it happens multiple times before I can even access the first passwords (on submitting my main password, then on accessing the credentials list). I have been waiting very impatiently for the 2024.9.0 update, and seeing that the fix hasn't made it to that update is incomprehensible to me, as it is for a number of others.

Self-hosters and people behind restrictive corporate firewalls may not be the majority of users, but maybe keep in mind that technologically savvy users tend to be loyal multiplicators for you as long as you don't antagonise them. Frankly, having to wait another three months without any comment on ETA feels like an undeserved level of communication for those who self-host, report bugs, and advocate Bitwarden to those around them.

**Comment by victornavorskie** [comment]
@kleinarne Sure, but for me it is really a problem, all my private passwords and work are in bitwarden and every day at work I have to contact in different services with sso and every time I have to connect to home with VPN to activate the bitwarden client. 
First I will install the @rchomat version if it works great, if not then maybe I will use keepass local for work until this issue is solved.

**Comment by popy2k14** [comment]
@justindbaur any news on this when we finally get a fixed client?

**Comment by rchomat** [comment]
Seems like the 2024.9.1 finally contains the fix but we need to wait several seconds every time we open the app or when you click on a tab (vault, send, generator, ...) when you want to access an item in your vault, ...
Better but still a painful usage...
@justindbaur please, find some time to improve that

**Comment by p-rintz** [comment]
> Seems like the 2024.9.1 finally contains the fix but we need to wait several seconds every time we open the app or when you click on a tab (vault, send, generator, ...) when you want to access an item in your vault, ... Better but still a painful usage... @justindbaur could you please, find some time to improve that ?

Cant confirm on Firefox.
Same issue as before.
Its the same as with 2024.9.0 where the vault unlocks, but the UI doesnt update until a while later or if you click out of the window and reopen Bitwarden. (as explained here: https://github.com/bitwarden/clients/issues/9561#issuecomment-2345512071 )

**Comment by trmartin4** [comment]
We are continuing to work on this issue and have some additional improvements in progress to make this experience better.  Thank you very much for your patience and for the feedback on exactly how the issue is occurring.

**Comment by farfromrefug** [comment]
I have the same issue on Firefox 132 with the extension 2024.11.0. Any chance that issue could be re open until this is fixed for everyone?

**Comment by popy2k14** [comment]
Same here. Better but definitely not fixed here (desktop and chromium extension).

@rchomat  @justindbaur please reopen.

Thx

**Comment by BetterB0y** [comment]
I'm still having an issue here.

**Comment by BetterB0y** [comment]
@rchomat @justindbaur Maybe, if there is an problem with solving it in better way, you will add option in settings to configure timeout value for connecting.

In `foreground-sync.service.ts` in line 85 there is a 30s timeout which is causing the issue. When I changed it to 1s it logged me instantly.

It's not ideal, but when database is already created usually I don't edit it often. So some kind of information that sync was not successful and option to change timeout would be great.

I didn't find it on desktop, but issue is similar.

**Comment by rchomat** [comment]
@BetterB0y I'm not part of bitwarden, just a user.
I don't have this issue anymore, at least I don't have to wait a lot. I use it on Chrome and Android. 
Are you sure you're using the la test version ?

Anyway, your suggestion could be a good way to  bypass

**Comment by BetterB0y** [comment]
I'm using the newest version - from chrome store. On MacOS desktop version issue is the same. I didn't check Windows desktop. I have self hosted server behind VPN. When i disconnect VPN there is 30s lag before db unlocks.

**Comment by rchomat** [comment]
When I unlock my DB with the master password, the ui is waiting for something but it's already unlocked. I usually quit the ui and click again on the icon. It opens fast

**Comment by BetterB0y** [comment]
You are right. DB is unlocking even if extension UI is lagging. But I feel like its not unlocking every time. Its pretty good, not ideal. I just haven't noticed that even if UI is lagged I can autofill already.

Still I think issue should be reopened again

**Comment by rchomat** [comment]
Maybe open a new issue ?

**Comment by BetterB0y** [comment]
Issue https://github.com/bitwarden/clients/issues/10790 was closed as duplicate of this. Maybe @Riqky can reopen?

**Comment by BetterB0y** [comment]
Its now resolved with version 2026.3.0
